llvm/utils/TableGen/Basic/SequenceToOffsetTable.h

```diff
@@ -127,10 +135,10 @@ class SequenceToOffsetTable {
   /// `\0`. Falls back to emitting a comma-separated integer list if
   /// `EmitLongStrLiterals` is false
   void emitStringLiteralDef(raw_ostream &OS, const Twine &Decl) const {
-    assert(Entries && "Call layout() before emitStringLiteralDef()");
+    assert(IsLaidOut && "Call layout() before emitStringLiteralDef()");
     if (!EmitLongStrLiterals) {
       OS << Decl << " = {\n";
-      emit(OS, printChar, "0");
+      emit(OS, printChar);
```

Comment:
Do we need to print the terminator here?

Reply:
It is printed by `emit` after each subsequence, assuming the terminator was provided.
The line below adds an extra '0' that I believe it should not do (and I think I ran into this before), but this is a pre-existing behavior and may be related to the issue I'm observing (see my comment below).

Reply:
The extra '\0' printing was added in https://reviews.llvm.org/D126136 and looks intentional. This way the generated data is the same regardless of EmitLongStrLiterals. If the option is enabled, the whole data is printed as a single string literal, and this string literal has an implicit terminating '\0' according to C. That change made the other case emit this extra '\0'.

It doesn't appear related to the issue with zero-sized arrays.